driver/src/orderbook/streamed/balance.rs

```diff
@@ -1,64 +1,108 @@
 use super::*;
-use error::Error;
+use anyhow::{anyhow, ensure, Result};
 use serde::{Deserialize, Serialize};
 
+/// The balance of a token for a user.
 #[derive(Clone, Copy, Debug, Default, Deserialize, Serialize)]
 pub struct Balance {
     balance: U256,
-    pub pending_deposit: Option<Flux>,
-    pub pending_withdraw: Option<Flux>,
+    deposit: Flux,
+    withdraw: Flux,
 }
 
 impl Balance {
-    pub fn deposit(&mut self, new_deposit: Flux, current_batch_id: BatchId) {
-        self.update_deposit_balance(current_batch_id);
-        match self.pending_deposit.as_mut() {
-            Some(deposit) => {
-                deposit.amount += new_deposit.amount;
-                deposit.batch_id = new_deposit.batch_id
-            }
-            None => self.pending_deposit = Some(new_deposit),
-        }
+    pub fn deposit(
+        &mut self,
+        amount: U256,
+        batch_id: BatchId,
+        current_batch_id: BatchId,
+    ) -> Result<()> {
+        ensure!(
+            batch_id == current_batch_id,
```

Comment:
Should this be `batch_id <= current_batch_id`? I guess it depends on how the event will be applied.

Reply:
The contract always sets the deposit's batch id to `getCurrentBatchId` so I think it's correct.

Reply:
Yes definitely! I meant if you were applying past deposits and your current batch was in the present, but if the events always get applied with the correct “current batch ID” then my point is moot.